stl_text/ops/tokenizers/whitespace_tokenizer.py

```diff
@@ -1,44 +1,62 @@
 from iopath.common.file_io import PathManager
-
+from torchtext.experimental.vocab import (
+    load_vocab_from_file,
+    Vocab,
+    vocab,
+)
 
 class WhitespaceTokenizer(nn.Module):
-    def __init__(self, vocab_path: Optional[str] = None, trainable=False, speed: int = 0):
+    r"""White space tokenizer and vocabulary builder
+
+    Args:
+        vocab_path Optional[str]: Path to vocabulary file. The file should contain tokens seperated by new line.
+        trainable (bool): Indicates whether to add tokens to dictionary on-the-fly
+
+    Example:
+
+        >>> from stl_text.ops.tokenizers import WhitespaceTokenizer
+        >>> whitespacetokenizer = WhitespaceTokenizer(trainable=True)
+        >>> jit_whitespacetokenizer = torch.jit.script(whitespacetokenizer)
+        >>> print(jit_whitespacetokenizer.code)
+        def forward(self,
+            text: str) -> List[int]:
+        _0 = (self).numberize(torch.split(text, None, -1), )
+        return _0
+        
+    """
+
+
+    def __init__(self, vocab_path: Optional[str] = None, trainable: bool = False):
         super(WhitespaceTokenizer, self).__init__()
         self.trainable = trainable
-        self.speed = speed  # mock a real tokenizer: slowing down tokenization speed
-
-        self.unknown = "unknown"
-        self.vocab: Dict[str, int] = {self.unknown: 0}
+        self.unk_token = '<unk>'
 
-        # load vocab
-        path_manager = PathManager()
         if vocab_path:
-            with path_manager.open(vocab_path, "r") as f:
-                for line in f.readlines():
-                    token = line.split()[0]
-                    self.vocab[token] = len(self.vocab)
+            path_manager = PathManager()
+            with path_manager.open(vocab_path, "r",encoding='utf-8') as f:
+                self.vocab = load_vocab_from_file(f,unk_token=self.unk_token)
+        else:
+            self.vocab = vocab(OrderedDict(),unk_token=self.unk_token)
+
+        self.vocab = self.vocab.to_ivalue() #Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
 
-    def forward(self, text: str) -> List[int]:
-        return self.numberize(self.tokenize(text))
 
-    def tokenize(self, text: str) -> List[str]:
-        # slowing down tokenization speed
-        count = 0
-        for i in range(self.speed):
-            count += 1
-        return text.split()
+    def forward(self, text: str) -> List[int]:
+        return self.numberize(text.split())
 
     def numberize(self, tokens: List[str]) -> List[int]:
-        token_ids: List[int] = []
-        for token in tokens:
-            if token in self.vocab:
-                token_ids.append(self.vocab[token])
-            else:
-                if self.trainable:
-                    token_ids.append(len(self.vocab))
-                    self.vocab[token] = len(self.vocab)
-                else:
-                    token_ids.append(self.vocab[self.unknown])
+        token_ids = self.vocab(tokens)
```

Comment:
Not very sure. Seem that we should separate  the `training` process from the `numberize` func. Probably we can have another func to train the vocab. @hudeven any special usage cases to couple these two steps together?

Reply:
@hudeven @zhangguanheng66 I included the build_vocab function inside whitespacetokenizer so that train vocab can be separated from forward function. On a separate note, I realize that https://github.com/hudeven/text/blob/master/examples/translation/train.py#L88 is failing on my side.